src/main/java/core/basesyntax/JavaDateTimeApi.java

```diff
@@ -110,16 +156,23 @@ public OffsetDateTime offsetDateTime(LocalDateTime localTime) {
      * return Optional of this date as a LocalDate.
      */
     public Optional<LocalDate> customParseDate(String date) {
-        return Optional.empty();
+        if (Integer.parseInt(date.substring(0, 2)) > 31) {
+            return Optional.empty();
+        }
+
+        return Optional.of(LocalDate.parse(date, DateTimeFormatter
+                .ofPattern("dd MMM yyyy").withLocale(Locale.UK)));
     }
 
     /**
      * Given some LocalDateTime, return a String formatted as
      * `day(2-digit) month(full name in English) year(4-digit) hours(24-hour format):minutes`.
-     *
+     * <p>
      * Example: "01 January 2000 18:00".
      */
     public String formatDate(LocalDateTime dateTime) {
-        return "";
+        DateTimeFormatter dateTimeFormatter = DateTimeFormatter.ofPattern("dd MMMM yyyy HH:mm")
```

Comment:
use local constant for patterns in the formatter

Reply:
done